File: backend/app/models/product.py
import decimal
import uuid

from sqlalchemy import Boolean, ForeignKey, Numeric, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, TimestampMixin, UUIDPrimaryKeyMixin


class Product(UUIDPrimaryKeyMixin, TimestampMixin, Base):
    __tablename__ = "products"

    gym_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("gyms.id", ondelete="CASCADE"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False)
    price: Mapped[decimal.Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    image_url: Mapped[str | None] = mapped_column(Text, nullable=True)
    in_stock: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True, server_default="true")

    gym = relationship("Gym", back_populates="products")

File: frontend/src/types/product.ts
export interface Product {
  id: string;
  gym_id: string;
  name: string;
  description: string;
  price: string;
  image_url: string | null;
  in_stock: boolean;
  created_at: string;
  updated_at: string;
}

export interface ProductPayload {
  name: string;
  description: string;
  price: number;
  image_url?: string | null;
  in_stock?: boolean;
}
